### ISSUE TITLE
Edges display

### ISSUE THREAD

**Issue by Peque** [bernhard-42/jupyter-cadquery#9]
It seems that convex edges are displayed/highlighted with a gray line, but concave edges are not.

In example, with this code the edges between the two extrusions are not displayed:

```python
from cadquery import Workplane

Workplane("XY").rect(2, 2).extrude(1).faces(">Z").rect(1, 1).extrude(1)
```

![Screenshot from 2020-01-15 22-33-57](https://user-images.githubusercontent.com/526577/72473717-954c4c80-37e7-11ea-81c4-99e2d03201f5.png)

Is that the expected behavior? How could it be fixed?

**Comment by bernhard-42** [comment]
@Peque Hmm, that is not nice indeed. I need to check whether I can control it in either cadquery, occ or pythreejs ....

**Comment by bernhard-42** [comment]
@Peque Luckily this can be solved, it's z-fighting and threejs has a config for that. You can test it like that:

```python
tower = cq.Workplane("XY").rect(2, 2).extrude(1).faces(">Z").rect(1, 1).extrude(1)

r = show(tower)

b = r.cq_view.pickable_objects.children[0]
b.material.polygonOffset = True
b.material.polygonOffsetUnits = 0.5
b.material.polygonOffsetFactor = 1
```

![image](https://user-images.githubusercontent.com/954408/72665330-9a87e200-3a07-11ea-8542-608df8e6b560.png)

I will add it into the next release

**Comment by bernhard-42** [comment]
@Peque Version 0.9.4 should include this fix.
Additionally I rewrote the viewing routine, since it did not work properly when the center of the bounding box was not (0,0,0). This also looks better now.

**Comment by Peque** [comment]
@bernhard-42 Thanks! I will try it today and report back. :blush:

**Comment by Peque** [comment]
Some feedback. :innocent: 

Edges
--------

The edges seem to be fixed for that simple example now. Thanks! :blush: 

It does not seem to work for different parts that intersect though:

```python
from cadquery import Workplane
from jupyter_cadquery.cadquery import Assembly
from jupyter_cadquery.cadquery import Part

tower1 = Workplane("XY").rect(2, 2).extrude(1)
tower2 = Workplane("XY").rect(1, 1).extrude(2)

Assembly([Part(tower1), Part(tower2)])
```

![Screenshot from 2020-01-18 19-00-25](https://user-images.githubusercontent.com/526577/72668219-d9786080-3a24-11ea-932d-12e4f3c15370.png)

Or is that expected?

Grid and axis
-----

I do not like the fact that the origin of the grid helper and axis helper is not at `(0, 0, 0)` now. In this example:

```python
Workplane("XY").rect(2, 2).extrude(1).faces(">Z").rect(1, 1).extrude(1)
```

I would expect the grid and the axis helpers to be bellow the tower, but they are crossing it:

![Screenshot from 2020-01-18 18-42-27](https://user-images.githubusercontent.com/526577/72668006-6b329e80-3a22-11ea-98c9-89b992b011c8.png)

**Comment by bernhard-42** [comment]
For the first issue: 
There are actually no edges from a model perspective. If you switch on transparency 
![image](https://user-images.githubusercontent.com/954408/72668319-cfa32d00-3a25-11ea-81dc-3ec65e8bddd6.png)

or rotate to the bottom:

![image](https://user-images.githubusercontent.com/954408/72668337-e6e21a80-3a25-11ea-839d-c643a7292b6c.png)

you can see the actual edges of the model. I don not modify the model in my previewer, I only show what I get. If you union the result `tower1.union(tower2)`, it is fine again:

![image](https://user-images.githubusercontent.com/954408/72668355-0b3df700-3a26-11ea-9f87-5889e794a3fd.png)

**Comment by bernhard-42** [comment]
For the second issue:
This is just the default setting where the axis are centered at the bounding box center of the object. This is btw. also the point where the model rotates around. 
To adapt, try:

```python
from jupyter_cadquery.cadquery import show
show(Workplane("XY").rect(2, 2).extrude(1).faces(">Z").rect(1, 1).extrude(1), 
     axes0=True, grid=True, axes=True, transparent=True)
```

which shows:

![image](https://user-images.githubusercontent.com/954408/72668409-b189fc80-3a26-11ea-9862-35a097a8882d.png)

Is this what you want?

**Comment by Peque** [comment]
Yeah, that solves all my issues. :tada:

I would still prefer the grid at `(0, 0, 0)` as the default behavior (feels more natural), but since I guess it is hard to do a survey and it is your project, you decide. :smile: 

Thanks a lot! :blush:

**Comment by bernhard-42** [comment]
How about:
```python
from jupyter_cadquery.cadquery import show as show_orig
def show(*shapes):
    show_orig(*shapes, axes0=True, grid=True, axes=True, transparent=True)
```
?

**Comment by Peque** [comment]
@bernhard-42 Yeah, it is more like a philosophical preference. :joy:  Not that is hard for me to patch it. But thanks! :heart:

**Comment by bernhard-42** [comment]
Got it. 
To be honest, I also don't know what I like more. I think I changed it because I played with some models that were far away from the origin and then I did not see grid or axis without zooming.

Let me play with it some more to find out my own preference  ...

**Comment by bernhard-42** [comment]
Hi @Peque 
I realised that I would love to have different defaults based on what I currently do. So I have added a kind of defaults management to jupyter-cadquery. You now can globally set your preferences:

```python
import cadquery as cq
from jupyter_cadquery.cadquery import Part, Assembly, show
from jupyter_cadquery import set_defaults, get_defaults, reset_defaults

set_defaults(axes=True, grid=True, axes0=True, transparent=True, cad_width=1000, position=(1,1,0))
```

Every following rendering will use these defaults. And you also can use `show(cad_objs, grid=False)` to overwrite the default for one specific plot.

It is available in version `0.9.5` and I hope that addresses your "philosophical preference" :wink:
